Core.ApplicationServices/LocalOptions/GenericLocalOptionsService.cs

```diff
@@ -0,0 +1,227 @@
+            _optionsRepository = optionsRepository;
+            _localOptionRepository = localOptionRepository;
+            _authorizationContext = authorizationContext;
+            _identityResolver = identityResolver;
+            _domainEvents = domainEvents;
+        }
+
+
+        public IEnumerable<TOptionType> GetByOrganizationUuid(Guid organizationUuid)
+        {
+            var globalOptions = GetGlobalOptionsAsQueryable()
+                .ToList();
+            
+            var localOptions = GetLocalOptionsByOrganizationUuid(organizationUuid)
+                .ToDictionary(x => x.OptionId);
+
+            return IncludeLocalChangesToGlobalOptions(globalOptions, localOptions);
+        }
+
+        private IEnumerable<TOptionType> IncludeLocalChangesToGlobalOptions(IEnumerable<TOptionType> globalOptions, IDictionary<int, TLocalOptionType> localOptions)
+        { 
+            return globalOptions.Select(optionToAdd =>
+              {
+                  var localOptionExists = localOptions.TryGetValue(optionToAdd.Id, out var localOption);
+                  if (localOptionExists) optionToAdd.UpdateLocalOptionValues(localOption);
+                  return optionToAdd;
+              });
+        }
+
+        public Result<TOptionType, OperationError> GetByOrganizationUuidAndOptionUuid(Guid organizationUuid, Guid globalOptionUuid)
+        {
+            return ResolveGlobalOptionId(globalOptionUuid)
+                .Bind(GetGlobalOptionById)
+                .Bind(option =>
+                {
+                    var localOption = GetLocalOptionsByOrganizationUuid(organizationUuid)
+                        .FirstOrDefault(x => x.OptionId == option.Id);
+
+                    option.UpdateLocalOptionValues(localOption);
+
+                    return Result<TOptionType, OperationError>.Success(option);
+                });
+        }
+
+        private Result<int, OperationError> ResolveGlobalOptionId(Guid uuid)
+        {
+            return _identityResolver.ResolveDbId<TOptionType>(uuid)
+                .Match(Result<int, OperationError>.Success,
+                    () => new OperationError($"Cannot resolve global option uuid {uuid} to database id", OperationFailure.NotFound));
+
+        }
+
+        public Result<TOptionType, OperationError> CreateLocalOption(Guid organizationUuid, LocalOptionCreateParameters parameters)
+        {
+            return ResolveOrganizationIdAndValidateCreate(organizationUuid)
+                .Bind(organizationId =>
+                {
+                    return GetLocalOptionByOrganizationUuidAndGlobalOptionUuid(organizationUuid, parameters.OptionUuid)
+                        .Match(existingLocalOption =>
+                        {
+                            existingLocalOption.Activate();
+                            _localOptionRepository.Update(existingLocalOption);
+                            _domainEvents.Raise(new EntityUpdatedEvent<TLocalOptionType>(existingLocalOption));
```

Comment:
It seems domain and application level functionality is bit mixed (at least for create/patch). Could maybe benefit from some separation.